release/nightly_tests/dataset/benchmark.py

```diff
@@ -0,0 +1,55 @@
+import gc
+import json
+import os
+import time
+from typing import Callable
+
+from ray.data.dataset import Dataset
+
+
+class Benchmark:
+    """Benchmark class used for Ray Datasets.
+
+    Call ``run(fn)`` to benchmark a specific piece of code/function.
+    Call ``write_result()`` to write benchmark result in file.
+    Result can be rendered in dashboard later through other tool.
+
+    A typical workflow would be:
+
+    benchmark = Benchmark(...)
+
+    # set up (such as input read or generation)
+    ...
+
+    benchmark.run(..., fn_1)
+    benchmark.run(..., fn_2)
+    benchmark.run(..., fn_3)
+
+    benchmark.write_result()
+
+    See example usage in ``aggregate_benchmark.py``.
+    """
+
+    def __init__(self, name):
+        self.name = name
+        self.result = {}
+        print(f"Running benchmark: {name}")
+
+    def run(self, name: str, fn: Callable[..., Dataset], **fn_run_args):
+        gc.collect()
+
+        print(f"Running case: {name}")
+        start_time = time.perf_counter()
+        output_ds = fn(**fn_run_args)
```

Comment:
We may want to run the benchmark multiple times to reduce noise. It's easy to add later to have a for-loop on this.

Reply:
Sounds good. This needs `fn` to be stateless/side-effect free.